### ISSUE TITLE
Report positions in error messages

### ISSUE THREAD

**Issue by julianhyde** [hydromatic/morel#118]
Report positions in compile-time and run-time error messages. The following examples are from sml-nj. Compile time:
```
fun f (true, true) = "a"
 | f (false, _) = "b"
 | f (_, true) = "c";
stdIn:2.1-7.21 Error: match redundant and nonexhaustive
          (true,true) => ...
          (false,_) => ...
    -->   (_,true) => ...
```
where "stIn:2.1-7.21" means "line 2 column 1 through line 7 column 21 of standard input".

Run time:
```
let
  fun f x y = x div (x - y)
in
  f 3 3
end;

uncaught exception Div [divide by zero]
  raised at: stdIn:2.17-2.20
```
We would probably implement this by baking the source region into a string literal that is the argument to the `div` function. If `div` throws, it would use the string literal in the exception.

**Comment by julianhyde** [comment]
Dev branch: https://github.com/julianhyde/morel/tree/0118-error-positions

**Comment by julianhyde** [comment]
Improved fix in 21c6fef.
